Build model and make migration for donations.

krsnamara opened the pull request:
Build out the relations between donor and donation.

krsnamara committed:
Build model and make migration for donations. Build out the relations between donor and donation.

sramalho94 reviewed (changes requested):
hey Martin, I'm doing a lap through all our models/migrations/controllers/routes. So sorry if some of this stuff isn't what you worked on for this PR but it's probably best if we fix everything in this PR right now, so:
- the Donor migration file is missing NOT NULL for some keys
- the Initiative migration file is missing NOT NULL for the title key
- Client model and migration files are missing NOT NULL for keys

we can work through everything together this morning

sramalho94 commented on controllers/DonationsController.js:
typo with Donation model name on ln 16 and 30

sramalho94 replied:
We'll have to make sure our get requests all include both the Donor model and iniative model

sramalho94 commented on migrations/20231206022357-create-donation.js:
initiative_id key is not properly configured for association between initiative and donation models

sramalho94 commented on models/donation.js:
Need to also establish association between donation and initiative models

sramalho94 committed:
Update DonationsController.js

Correct typo in Donation model name
Add include statement in getDonationById controller

sramalho94 committed:
Update 20231205215252-create-donors.js

Add allowNull restrictions to keys

sramalho94 committed:
Update 20231205232854-create-initiative.js

Add allowNull restriction to title key

sramalho94 committed:
Update 20231205233905-create-client.js

Add notNull restrictions to keys

sramalho94 committed:
Update client.js

Add notNull restrictions to keys
